openzl-util/src/codec.rs

```diff
@@ -850,6 +853,23 @@ where
     }
 }
 
+#[cfg(feature = "alloc")]
+#[cfg_attr(doc_cfg, doc(cfg(feature = "alloc")))]
+impl<T> Decode for Box<[T]>
+where
+    T: Decode,
+{
+    type Error = Option<T::Error>;
```

Comment:
I don't know, `Vec<T>` and `[T; N]` have the same error type

Reply:
One of the error possibilities is that the vector itself is not well-formed in which case we use `None` otherwise there could be an error in an individual element so we have `T::Error`. A better type perhaps would be a custom enum.